mesh/nurbs.cpp

```diff
@@ -574,6 +829,37 @@ void KnotVector::FindInterpolant(Array<Vector*> &x)
    }
 }
 
+// Routine from "The NURBS book" - 2nd ed - Piegl and Tiller
+// Algorithm A9.1 p. 369
+void KnotVector::GetInterpolant(const Vector &x, const Vector &u,
+                                Vector &a) const
+{
+   int ks;
+   real_t xi;
+
+   // Assemble collocation matrix
+   Vector shape(Order+1);
+   DenseMatrix A(NumOfControlPoints,NumOfControlPoints);
```

Comment:
For a square matrix, the size can be given just once.

Reply:
Simplified.